x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_tag_badge.tsx

```diff
@@ -34,32 +40,31 @@ const getTagItemDataTestSubj = (tag: string) => `ruleTagBadgeItem-${tag}`;
 
 export const RuleTagBadge = (props: RuleTagBadgeProps) => {
   const {
-    isOpen = false,
     tags = [],
-    onClick,
-    onClose,
     badgeDataTestSubj = 'ruleTagBadge',
     titleDataTestSubj = 'ruleTagPopoverTitle',
     tagItemDataTestSubj = getTagItemDataTestSubj,
   } = props;
 
   const badge = useMemo(() => {
-    return (
-      <EuiBadge
-        data-test-subj={badgeDataTestSubj}
-        color="hollow"
-        iconType="tag"
-        iconSide="left"
-        tabIndex={-1}
-        onClick={onClick}
-        onClickAriaLabel="Tags"
-        iconOnClick={onClick}
-        iconOnClickAriaLabel="Tags"
-      >
-        {tags.length}
-      </EuiBadge>
-    );
-  }, [tags, badgeDataTestSubj, onClick]);
+    if ('onClick' in props) {
```

Comment:
Would it be better to use `spread` as the flag to determine which "mode" this component is in? using onClick seems a bit flaky because, for example, what if we wanted to add a click handler to the spread tags in the future?